src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -926,6 +933,28 @@ private Conversion ClassifyImplicitBuiltInConversionFromExpression(BoundExpressi
             return Conversion.NoConversion;
         }
 
+        private Conversion GetSwitchExpressionConversion(BoundExpression source, TypeSymbol destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        {
+            switch (source)
+            {
+                case BoundConvertedSwitchExpression _:
+                    // It has already been subjected to a switch expression conversion.
+                    return Conversion.NoConversion;
```

Comment:
>NoConversion [](start = 38, length = 12)

can we reach this?

Reply:
Almost certainly, especially during error recovery and error reporting.

---
In reply to: [301774871](https://github.com/dotnet/roslyn/pull/37052#discussion_r301774871) [](ancestors = 301774871)

Reply:
Though I have not found any tests that fail with a throw statement here.

---
In reply to: [302304419](https://github.com/dotnet/roslyn/pull/37052#discussion_r302304419) [](ancestors = 302304419,301774871)